CMakeLists.txt

```diff
@@ -212,6 +212,18 @@ if (${SPIRV_CHECK_CONTEXT})
   add_definitions(-DSPIRV_CHECK_CONTEXT)
 endif()
 
+# Precompiled header macro. Parameters are source file list and filename for pch cpp file.
+macro(PCH SRCS PCHCPP)
+  if(MSVC AND CMAKE_GENERATOR MATCHES "^Visual Studio")
+    set(PCH_NAME "$(IntDir)\\pch.pch")
```

Comment:
Is it possible that you will be writing to the same `pch.pch` for each of the `pch.cpp` files you have?  Another reason to have different names.

Reply:
In Visual Studio, each project's pch.pch goes to a unique output directory. That doesn't happen with ninja, and was the primary source of problems with that build.